autocorpus/bioc_passages.py

```diff
@@ -1,70 +1,65 @@
 """BioC Passage builder script."""
 
+from __future__ import annotations
 
+from dataclasses import asdict, dataclass
+from typing import Any
+
+_DEFAULT_KEYS = set(("section_heading", "subsection_heading", "body", "section_type"))
+
+
+@dataclass(frozen=True)
 class BioCPassage:
-    """BioC Passage builder class."""
+    """Represents a BioC passage."""
+
+    offset: int
+    infons: dict[str, Any]
+    text: str
 
     @classmethod
-    def from_title(cls, title, offset):
-        """Creates a BioCPassage object from a title.
+    def from_dict(cls, passage: dict[str, Any], offset: int) -> BioCPassage:
```

Comment:
Maybe it is better to make this a standalone function? I'm not sure, there is a certain elegance to having it this way, and it is maybe more readable when it is called because you know the type of the variable just by looking at the code.

Reply:
Like you say, I think it's a nice way to show the return type. If it was returning a plain old `dict` I think it might be different. Plus there is a separate `from_title` classmethod and I think it makes sense to keep them as a pair.